be/src/exec/stream/aggregate/agg_state_data.cpp

```diff
@@ -0,0 +1,294 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exec/stream/aggregate/agg_state_data.h"
+
+#include "exprs/agg/stream/stream_detail_state.h"
+
+namespace starrocks::stream {
+
+Status AggStateData::allocate_intermediate_state(size_t chunk_size, const std::vector<uint8_t>& keys_not_in_map,
+                                                 const std::vector<ChunkPtrOr>& result_chunks,
+                                                 const Buffer<AggGroupStatePtr>& agg_group_state) const {
+    int32_t j = 0;
+    auto table_idx = intermediate_table_idx();
```

Comment:
here table_idx is column index into chunk, so suggest to use column_idx and intermediate_column_idx

Reply:
fixed.